versioned/dynamodb/src/main/java/com/dremio/nessie/versioned/implrocksdb/RocksDBStoreManager.java

```diff
@@ -0,0 +1,278 @@
+      add(VALUE_COLUMN_FAMILY);
+      add(KEY_FRAGMENT_COLUMN_FAMILY);
+      add(COMMIT_METADATA_AUTHOR_COLUMN_FAMILY);
+      add(COMMIT_METADATA_MESSAGE_COLUMN_FAMILY);
+    }};
+  private static final long ROCKSDB_OPEN_SLEEP_MILLIS = 100L;
+  private final String baseDirectory;
+
+  private RocksDB db;
+  private ColumnFamilyHandle defaultHandle;
+  // the Options class contains a set of configurable DB options
+  // that determines the behaviour of the database.
+  // private Options options;
+  private boolean noDBOpenRetry;
+  // on #start all the existing tables are loaded, and if new stores are requested, #newStore is used
+  private final ConcurrentMap<Integer, String> handleIdToNameMap = Maps.newConcurrentMap();
+  private final LoadingCache<String, RocksDBStore> maps = CacheBuilder.newBuilder()
+      .removalListener((RemovalListener<String, RocksDBStore>) notification -> {
+        try {
+          notification.getValue().close();
+        } catch (Exception ex) {
+          //TODO add exception
+          //closeException.addException(ex);
+        }
+      }).build(new CacheLoader<String, RocksDBStore>() {
+        @Override
+        public RocksDBStore load(String name) throws RocksDBException {
+          return newStore(name);
+        }
+      });
+
+  public RocksDBStoreManager(String baseDirectory, boolean noDBOpenRetry) {
+    this.baseDirectory = baseDirectory;
+    this.noDBOpenRetry = noDBOpenRetry;
+  }
+
+  private RocksDBStore newStore(String name) throws RocksDBException {
+    final ColumnFamilyDescriptor columnFamilyDescriptor = new ColumnFamilyDescriptor(name.getBytes(UTF_8));
+    ColumnFamilyHandle handle = db.createColumnFamily(columnFamilyDescriptor);
+    handleIdToNameMap.put(handle.getID(), name);
+
+    return newRocksDBStore(name, columnFamilyDescriptor, handle);
+  }
+
+  private RocksDBStore newRocksDBStore(String name, ColumnFamilyDescriptor columnFamilyDescriptor,
+                                       ColumnFamilyHandle handle) {
+    /*
+    final RocksMetaManager rocksManager;
+    if (BLOB_WHITELIST.contains(name)) {
+      rocksManager = new RocksMetaManager(baseDirectory, name, FILTER_SIZE_IN_BYTES);
+    } else {
+      rocksManager = new RocksMetaManager(baseDirectory, name, Long.MAX_VALUE);
+    }
+    */
+    return new RocksDBStore(name, columnFamilyDescriptor, handle, db);
+  }
+
+  /**
+   * This method opens the RocksDB database. It creates a cache of the tables(Column Families) created.
+   */
+  @Override
+  public void start() {
+    //TODO need to make path configurable.
+    List<byte[]> families = null;
+
+    //TODO implement valiation of path existing
+
+    try (final Options options = new Options()) {
+      options.setCreateIfMissing(true);
+      // get a list of existing families.
+      families = new ArrayList<>(RocksDB.listColumnFamilies(options, baseDirectory));
+    } catch (RocksDBException e) {
+      LOGGER.error("Error extracting RocksDB options", e);
```

Comment:
Seems like you should be propagating the exceptions?

Reply:
Agreed. Will change.